woeip/apps/air_quality/views.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import Error
+from django.shortcuts import redirect, render
+from django.utils.encoding import force_text
+
+from woeip.apps.air_quality import dustrak, forms, models
+
+logger = logging.getLogger(__name__)
+
+
+def index(request):
+    return render(request, 'index.html')
+
+
+@login_required
+def upload(request):
+    """Upload data for a session collected using the Dustrak air quality device and a separate GPS
+    log file. Creates a new Session instance, two SessionData instances, and Data instances for
+    each sample.
+    """
+
+    if request.method == 'POST':
+        form = forms.DustrakSessionForm(request.POST, request.FILES)
+        if form.is_valid():
+            form.save()
+            try:
+                air_sensor = models.Sensor.objects.get(name='Dustrak')
+                gps_sensor = models.Sensor.objects.get(name='GPS')
+
+                air_quality = models.SessionData(upload=request.FILES['air_quality'],
+                                                 sensor=air_sensor,
+                                                 session=form.instance,
+                                                 uploaded_by=request.user)
+                gps = models.SessionData(upload=request.FILES['gps'],
+                                         sensor=gps_sensor,
+                                         session=form.instance,
+                                         uploaded_by=request.user)
+
+                air_quality_contents = force_text(request.FILES['air_quality'].read())
+                _, air_quality_data = dustrak.load_dustrak(air_quality_contents, 'America/Los_Angeles')
+
+                gps_contents = force_text(request.FILES['gps'].read())
+                gps_data = dustrak.load_gps(gps_contents)
+                joined_data = dustrak.join(air_quality_data, gps_data)
+
+            except (UnboundLocalError, ObjectDoesNotExist, Error) as e:
+                messages.add_message(request, messages.ERROR, f'File upload failed, error: {e}')
```

Comment:
Add a logging line—`log.exception()`—so that data is written to the log files. `exception()` has the benefit of including the stack trace, so no need to pass the exception.

Reply:
From the `logging` documentation, project code, and error messages, it seems the exact syntax should be `logger.exception(e)` or `logging.exception(e)`. Unfortunately, I was unable to confirm this because I also could not find the log file. From reading `settings.py`, shouldn't it be towards the base directory of the project?

Reply:
Logs are written to the `logs` directory at the repo root, assuming `ENABLE_LOGGING_TO_FILE` is `True`. Otherwise, logs are only written to the console.

https://github.com/openoakland/woeip/blob/master/woeip/settings.py#L187

Reply:
It seems my local system objected to Django creating a file and writing to it, while woeip was running in a Docker Container. I changed [Line 202](https://github.com/openoakland/woeip/blob/master/woeip/settings.py#L202) to `/app/woeip/app.log`, to follow a path used by the container.

I imagine that someone running woeip directly on a local system (not using Docker) would not encounter this issue. For them, instead of fixing an issue, my addition of `/app/woeip/` would bury the log file within two folders.

Reply:
Exactly how did your local system object? What did you set? What was the resulting output?

The code should work in all cases. If that is not true, it needs to be fixed.

Reply:
* Within `docker-compose.local.yml`, under `app` -> `environment`, I created the variable `ENABLE_LOGGING_TO_FILE` and set it to `True`
* I haven't been receiving errors in the terminal and can't provide one in this case, either.
* The expected behavior was that an app.log file would be created and populated with a known error
* The observed behavior was that no such file was created
* I modified `settings.py`, as described above. This created the expected behavior.